rest/route/project.go

```diff
@@ -379,6 +379,23 @@ func (h *projectIDPatchHandler) Run(ctx context.Context) gimlet.Responder {
 	}
 
 	newRevision := model.FromStringPtr(requestProjectRef.Revision)
+	// user didn't provide updated revision, but we should still update if branch has changed
+	if newRevision == "" && newProjectRef.Branch != oldProject.Branch && !newProjectRef.HasRepotrackerError() {
```

Comment:
If the user hasn't provided a new revision, I would expect that running the repotracker would fail. In this case, we are only running the repotracker in order to set the error. However, we don't know how long it will take to get the error, and I think you're right that it's awkward to run a job in a route.

Would it make sense instead to run a job only in the `newRevision != ""` case, but run it asynchronously (add it to the queue, don't get the error). 

We could document that the user should probably update the new revision if they're switching the branch. Or (this seems slightly odd, but I haven't been able to talk myself out of it) we could add validation that if you're switching a branch, you should also provide a new revision.